qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/PersonSteps.java

```diff
@@ -0,0 +1,115 @@
+	
+    private AdministrationPage administrationPage;
+    private ManagePersonPage managePersonPage;
+    private PersonFormPage personFormPage;
+    private AddPersonPage addPersonPage;
+    private String personUuid;
+    private TestData.PersonInfo personInfo;
+    private final String PERSON_NAME = RandomStringUtils.randomAlphabetic(8);
+    private final String PERSON_AGE = RandomStringUtils.randomNumeric(2);
+    private final String PERSON_FAMILY_NAME = RandomStringUtils.randomAlphanumeric(8);
+	
+    @Before(RunTest.HOOK.SELENIUM_PERSON)
+    public void setup() {
+	initiateWithLogin();
+	personInfo = TestData.generateRandomPerson();
+	personUuid = TestData.createPerson(personInfo);
+    }
+	
+    @After(RunTest.HOOK.SELENIUM_PERSON)
+    public void teardown() {
+	RestClient.delete("person/" + personUuid, true);
+	quit();
+    }
+	
+    @Given ("User clicks on the system administration link on the home page")
+    public void launchSystemAdministrationPage() {
+    	administrationPage = homePage.goToAdministration();
+    }
+    
+    @When ("User clicks on the manage Persons link on the administration page")
+    public void launchAdvancedAdministrationPage() {
+    	managePersonPage = administrationPage.clickOnManagePersons();
+    }
+    
+    @And ("User clicks on the create person link on the manage persons page")
+    public void clickOnCreatePersonPage() {
+    	addPersonPage = managePersonPage.createPerson();
+    	addPersonPage.createPerson();
+    }
+
+    @Then ("User fills the create person form")
+    public void createPersonForm() {
+    	addPersonPage.setPersonName(PERSON_NAME);
+    	addPersonPage.setAge(PERSON_AGE);
+    	addPersonPage.clickGenderMale();
+    	personFormPage = addPersonPage.createPerson();
+    	personFormPage.setFamilyNameField(PERSON_FAMILY_NAME);
+    	personFormPage.savePerson();
+    	personFormPage.deletePersonForever();
+    }
+    
+    @And ("User searches for the person")
+    public void searchForPersonToEdit() {
+    	managePersonPage.setPersonName(personInfo.givenName);
+    }
+    
+    @And ("User clicks on the first person found from the search results")
+    public void clickFirstPersonFound() {
+    	personFormPage = managePersonPage.clickFirstFoundPerson();
+    }
+    
+    @And ("User edits the person details")
+    public void editPersonDetails() {
+    	personFormPage.setFamilyNameField("newFamilyName");
+    	personFormPage.savePerson();
+    	assertThat(personFormPage.getFamilyName(), is("newFamilyName"));
+    }
+    
+    @Then ("User clicks on the delete person forever button")
+    public void deletePersonForever() {
+    	personFormPage.deletePersonForever();
+    }
+    
+    @And ("User fills in the reason for retiring person")
+    public void setRetireReason() {
+    	personFormPage.setRetireReason("retire reason");
+    }
+    
+    @Then ("User clicks on the retire person button")
+    public void retirePerson() {
+    	personFormPage.retirePerson();
+    }
+}
+
```

Comment:
isn't one empty line at the EOF sufficient enough to cater for git collaborators?